src/main/scala/org/scalajs/tools/tsimporter/parser/TSDefParser.scala

```diff
@@ -276,13 +276,14 @@ class TSDefParser extends StdTokenParsers with ImplicitConversions {
   lazy val indexMember: Parser[MemberTree] =
     ("[" ~> identifier ~ typeAnnotation <~ "]") ~ typeAnnotation ^^ IndexMember
 
-  lazy val namedMember: Parser[MemberTree] =
+  lazy val namedMember: Parser[MemberTree] = {
     modifiers ~ propertyName ~ optionalMarker >> {
       case mods ~ name ~ optional => (
           functionSignature ^^ (FunctionMember(name, optional, _, mods))
         | typeAnnotation ^^ (PropertyMember(name, optional, _, mods))
       )
-    }
+    } | "private" ~> propertyName ~ (functionSignature | ";") ^^^ PrivateMember
```

Comment:
The semi-colon must not be there. It is already present in the definition of `memberBlock`.

I suggest extracting this case in a separate `lazy val privateMember`, and add `privateMember` as an alternative to `typeMember` rather than to `namedMember`.

Reply:
Done